LiteCore/Support/LogEncoder.cc

```diff
@@ -95,14 +94,13 @@ namespace litecore {
         _writer.write(&_level, sizeof(_level));
         _writeStringToken(domain ? domain : "");
 
-        _writeUVarInt((unsigned)object);
-        if (object != ObjectRef::None) {
-            auto i = _objects.find(unsigned(object));
-            if (i != _objects.end() && i->second != "\x1") {
-                _writer.write(slice(i->second));
-                _writer.write("\0", 1);
-                i->second = string("\x1"); // Remove this from memory, but keep so we can track as "seen"
-            }
+        const auto objRef = (unsigned)object;
+        _writeUVarInt(objRef);
+        if (object != ObjectRef::None && _seenObjects.find(objRef) == _seenObjects.end()) {
+            _seenObjects.insert(objRef);
+            auto i = LogDomain::getObject(objRef);
```

Comment:
`name` would be a clearer variable name. (I just called it `i` before because it was an iterator)